src/fortrancode.l

```diff
@@ -732,6 +733,54 @@ LANGUAGE_BIND_SPEC BIND{BS}"("{BS}C{BS}(,{BS}NAME{BS}"="{BS}"\""(.*)"\""{BS})?")
+                                            char *tmp= (char *) malloc(strlen(yytext)+1);
+                                            strcpy(tmp,yytext);
+                                            strtok(tmp," \t(");
+                                            //printf("     generateLink |%s|\n",tmp);
+                                            generateLink(*g_code, tmp);
+                                            if ( strlen(tmp) != strlen(yytext) ) {
+                                              // everything else is thrown back into input stream
+                                              // to prevent {BS} from being gobbled
+                                              //printf("     reprocess |%s|\n",yytext+strlen(tmp));
+                                              yyless(strlen(tmp));
+                                            }
+                                            free(tmp);
+					    g_insideBody=FALSE;
+                                        }
+
+ /* array(.)= assignment is hard to catch.                                      */
+ /* (.) could contain other variables, structures, function calls, etc.         */
+ /* <Start>{ID}{BS}/"("                                                         */
+ /*   Too Simple. Catches both "variable(.)=" and "keyword(.)" [ie, "case(1)"]. */
+ /* <Start>{ID}{BS}/"("[a-z_A-Z0-9 \t,:\-+/*%()]+")"{BS}"="                     */
+ /*   Error:  Dangerous trailing context (due to varying length?):              */
+ /* Method below best fix so far, but might not be perfect.                     */
+ /*   Keeps most assignments from being detected as keywords.                   */
+ /*   What is missed, gets processed later as a keyword or variable reference.  */
+
+<Start>{ID}{BS}"("[a-z_A-Z0-9 \t\.,:+*\-/%()]+")"{BS}"="{BS}/[^=] { // array(.) assignment
+                                            // yytext matches "array(.)="
+                                            //printf("===> line number %d array assignment |%s|\n",g_yyLineNr,yytext);
+  					    g_insideBody=TRUE;
+                                            // extract array name and process it
+                                            char *tmp= (char *) malloc(strlen(yytext)+1);
+                                            strcpy(tmp,yytext);
+                                            strtok(tmp," \t(");
+                                            //printf("     generateLink |%s|\n",tmp);
+                                            generateLink(*g_code, tmp);
+                                            // everything else is thrown back into input stream
+                                            //printf("     reprocess |%s|\n",yytext+strlen(tmp));
+                                            yyless(strlen(tmp));
+                                            free(tmp);
+					    g_insideBody=FALSE;
+                                        }
```

Comment:
Why not end the "search at the `(` like with a scalar variable so that the indices can be handled (and linked when necessary) separately
How is currently a multiline array handled, i.e. when we have a continuation line in the indices of the array?

Reply:
On line 768, ending the search at ( would not discriminate from certain fortran keywords, like "real(int_expression)" or "case(value)".  Placing / immediately after {ID} and treating the rest as trailing context generates the "dangerous trailing context" compiler warning.  So, the proposed context captures everything to the "=" and then uses the "(" in strtok on line 775 to push the indices back onto the stack so that links can be generated.

This construct will not capture array assignment if line continuation occurs before the = operator.  We rely upon the simple variable reference lexer of line 1281 to catch those.